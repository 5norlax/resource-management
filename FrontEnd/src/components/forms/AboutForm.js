import React from 'react';
import { Label, Input, FormGroup, Row } from 'reactstrap';

const AboutForm = ({ FirstName, LastName, Role, Email, Phone, LinkedIn, GitHub, PersonalSite }) => {
    return (
        <div>
            <Row>
                <FormGroup className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12">
                    <Label for="firstName">First Name</Label>
                    <Input type="text" name="firstName" id="firstName" placeholder="First Name" value={FirstName} />
                </FormGroup>
                <FormGroup className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12">
                    <Label for="lastName">Last Name</Label>
                    <Input type="text" name="lastName" id="LastName" placeholder="Last Name" value={LastName} />
                </FormGroup>
                <FormGroup className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12">
                    <Label for="role">Role</Label>
                    <Input type="text" name="role" id="role" placeholder="Title/Role" value={Role} />
                </FormGroup>
                <FormGroup className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12">
                    <Label for="email">Email</Label>
                    <Input type="email" name="email" id="email" placeholder="Email Address" value={Email} />
                </FormGroup>
                <FormGroup className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12">
                    <Label for="phone">Phone</Label>
                    <Input type="text" name="phone" id="phone" placeholder="Phone Number" value={Phone} />
                </FormGroup>
                <FormGroup className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12">
                    <Label for="linkedIn">LinkedIn</Label>
                    <Input type="text" name="linkedIn" id="linkedIn" placeholder="linkedin.com/in/yourlink (Optional)" value={LinkedIn} />
                </FormGroup>
                <FormGroup className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12">
                    <Label for="github">GitHub</Label>
                    <Input type="text" name="github" id="github" placeholder="github.com/youraccount (Optional)" value={GitHub} />
                </FormGroup>
                <FormGroup className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12">
                    <Label for="personalSite">Personal Website</Label>
                    <Input type="text" name="personalSite" id="personalSite" placeholder="www.yoursiteurl.com (Optional)" value={PersonalSite} />
                </FormGroup>
            </Row>
            <hr />
        </div>
    )
}

export default AboutForm