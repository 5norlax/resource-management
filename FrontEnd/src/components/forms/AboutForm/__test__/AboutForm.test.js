import React from 'react';
import { render, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import AboutForm from '../AboutForm';

afterEach(cleanup);

describe('Test AboutForm', () => {

  it('matches snapshot', () => {
    const tree = renderer.create(<AboutForm />).toJSON()
    expect(tree).toMatchSnapshot();
  });

});