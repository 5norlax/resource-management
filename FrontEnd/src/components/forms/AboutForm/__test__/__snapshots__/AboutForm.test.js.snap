// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test AboutForm matches snapshot 1`] = `
<div>
  <div
    className="row"
  >
    <div
      className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12 form-group"
    >
      <label
        className=""
        htmlFor="FirstName"
      >
        First Name
      </label>
      <input
        autoFocus={false}
        className="form-control"
        id="FirstName"
        name="FirstName"
        placeholder="First Name"
        required={true}
        spellCheck="false"
        type="text"
      />
    </div>
    <div
      className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12 form-group"
    >
      <label
        className=""
        htmlFor="LastName"
      >
        Last Name
      </label>
      <input
        className="form-control"
        id="LastName"
        name="LastName"
        placeholder="Last Name"
        required={true}
        spellCheck="false"
        type="text"
      />
    </div>
    <div
      className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12 form-group"
    >
      <label
        className=""
        htmlFor="Role"
      >
        Role
      </label>
      <select
        className="form-control"
        id="Role"
        name="Role"
        placeholder="Title/Role"
        required={true}
      >
        <option
          disabled={true}
          hidden={true}
          selected={true}
          value=""
        >
          Please Select
        </option>
        <option>
          Business Analyst
        </option>
        <option>
          Data Analyst
        </option>
        <option>
          DevOps Engineer
        </option>
        <option>
          Jr. Software Developer
        </option>
        <option>
          Manager
        </option>
        <option>
          Software Engineer
        </option>
        <option>
          Solutions Architect
        </option>
        <option>
          Sr. Technical Lead
        </option>
        <option>
          UI/UX Developer
        </option>
      </select>
    </div>
    <div
      className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12 form-group"
    >
      <label
        className=""
        htmlFor="Email"
      >
        Email
      </label>
      <input
        className="form-control"
        id="Email"
        name="Email"
        placeholder="Email Address"
        required={true}
        type="email"
      />
    </div>
    <div
      className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12 form-group"
    >
      <label
        className=""
        htmlFor="Phone"
      >
        Phone
      </label>
      <input
        className="form-control"
        id="Phone"
        name="Phone"
        pattern="^[0-9]*$"
        placeholder="Phone Number (##########)"
        required={true}
        type="tel"
      />
    </div>
    <div
      className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12 form-group"
    >
      <label
        className=""
        htmlFor="LinkedIn"
      >
        LinkedIn
      </label>
      <input
        className="form-control"
        id="LinkedIn"
        name="LinkedIn"
        placeholder="https://www.linkedin.com/in/yourlink (Optional)"
        type="url"
      />
    </div>
    <div
      className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12 form-group"
    >
      <label
        className=""
        htmlFor="GitHub"
      >
        GitHub
      </label>
      <input
        className="form-control"
        id="GitHub"
        name="GitHub"
        placeholder="https://www.github.com/youraccount (Optional)"
        type="url"
      />
    </div>
    <div
      className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12 form-group"
    >
      <label
        className=""
        htmlFor="PersonalSite"
      >
        Personal Website
      </label>
      <input
        className="form-control"
        id="PersonalSite"
        name="PersonalSite"
        placeholder="https://www.yoursiteurl.com (Optional)"
        type="url"
      />
    </div>
  </div>
  <hr />
</div>
`;
